fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/PruneOlapScanPartition.java

```diff
@@ -0,0 +1,166 @@
+            return ctx.root;
+        }).toRule(RuleType.PARTITION_PRUNE);
+    }
+
+    private ColumnRange createColumnRange(String colName, List<Expression> expressionList) {
+        ColumnRange result = ColumnRange.create();
+        for (Expression expression : expressionList) {
+            Expression leftMostChild = expression.leftMostNode();
+            if (!(leftMostChild instanceof SlotReference)) {
+                continue;
+            }
+            SlotReference slotRef = (SlotReference) leftMostChild;
+            if (!slotRef.boundToColumn(colName)) {
+                continue;
+            }
+            if (expression instanceof Or) {
+                ColumnRanges ranges = exprToRanges(expression, colName);
+                switch (ranges.type) {
+                    case IS_NULL:
+                        result.setHasConjunctiveIsNull(true);
+                        break;
+                    case CONVERT_SUCCESS:
+                        result.intersect(ranges.ranges);
+                        break;
+                    case CONVERT_FAILURE:
+                    default:
+                        break;
+                }
+            } else {
+                ColumnRanges ranges = exprToRanges(expression, colName);
+                switch (ranges.type) {
+                    case IS_NULL:
+                        result.setHasConjunctiveIsNull(true);
+                        break;
+                    case CONVERT_SUCCESS:
+                        result.intersect(ranges.ranges);
+                        break;
+                    case CONVERT_FAILURE:
+                    default:
+                        break;
+                }
+            }
+        }
+        return result;
+    }
+
+    private ColumnRanges exprToRanges(Expression expression, String colName) {
+        // TODO: process in/is null expression
+        if (!(expression instanceof ComparisonPredicate)) {
+            return ColumnRanges.createFailure();
+        }
+        List<Range<ColumnBound>> result = Lists.newArrayList();
+        ComparisonPredicate comparisonPredicate = (ComparisonPredicate) expression;
+        Expression rightChild = comparisonPredicate.child(1);
+        if (rightChild == null || !rightChild.isConstant() || !(rightChild instanceof Literal)) {
+            return ColumnRanges.createFailure();
+        }
+        LiteralExpr value = ((Literal) rightChild).toLegacyLiteral();
+        if (expression instanceof EqualTo) {
+            ColumnBound bound = ColumnBound.of(value);
+            result.add(Range.closed(bound, bound));
+        } else if (expression instanceof GreaterThanEqual) {
+            result.add(Range.atLeast(ColumnBound.of(value)));
+        } else if (expression instanceof GreaterThan) {
+            result.add(Range.greaterThan(ColumnBound.of(value)));
+        } else if (expression instanceof LessThan) {
+            result.add(Range.atMost(ColumnBound.of(value)));
```

Comment:
```suggestion
            result.add(Range.LessThan(ColumnBound.of(value)));
```

add more test please